service-base/src/main/java/org/eclipse/hono/service/monitoring/AbstractMessageSenderConnectionEventProducer.java

```diff
@@ -82,27 +85,45 @@ protected AbstractMessageSenderConnectionEventProducer(
             return Future.succeededFuture();
         }
 
-        return getOrCreateSender(context.getMessageSenderClient(), authenticatedDevice.getTenantId())
-                .compose(sender -> {
-
-                    final JsonObject payload = new JsonObject();
-                    payload.put("cause", cause);
-                    payload.put("remote-id", remoteId);
-                    payload.put("source", protocolAdapter);
-
-                    if (data != null) {
-                        payload.put("data", data);
-                    }
-
-                    return sender.send(
-                            authenticatedDevice.getDeviceId(),
-                            payload.encode().getBytes(StandardCharsets.UTF_8),
-                            EventConstants.EVENT_CONNECTION_NOTIFICATION_CONTENT_TYPE
-                            );
+        return context.getTenantClientFactory().getOrCreateTenantClient()
+                .map(tenantClient -> tenantClient.get(authenticatedDevice.getTenantId()))
+                .map(tenantObjectFuture -> {
+                    return getOrCreateSender(context.getMessageSenderClient(), authenticatedDevice.getTenantId())
+                            .compose(sender -> {
+
+                                final JsonObject payload = new JsonObject();
+                                payload.put("cause", cause);
+                                payload.put("remote-id", remoteId);
+                                // TODO: can this be removed in favour of 'orig_dapter'?
```

Comment:
I don't think so because the Event API defines this as a mandatory property of the message body.